### PR TITLE
Implementing SignInScreen viewmodel, logic and listeners.

### PR DESCRIPTION
**Pull request by bohdanzdaniekin**

### CODE REVIEW

File: app/src/main/java/com/mr/nemo/dragonfly/domain/entity/auth/signup/PasswordValue.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.mr.nemo.dragonfly.domain.entity.auth.signup
+
+data class PasswordValue(
+    val password: String = "",
+    val confirmation: String = ""
+)
```
Comment (slothdeboss) [inline app/src/main/java/com/mr/nemo/dragonfly/domain/entity/auth/signup/PasswordValue.kt:6]: @bohdanzdaniekin is it a good idea to have `confirmation` here too?

Comment (bohdanzdaniekin) [reply app/src/main/java/com/mr/nemo/dragonfly/domain/entity/auth/signup/PasswordValue.kt:6]: image
these values are tightly coupled, so I'm not sure I should separate them into different classes/properties.

File: app/src/main/java/com/mr/nemo/dragonfly/ui/screen/auth/signin/SignInScreen.kt [hunk 2]
```diff
@@ -171,23 +219,38 @@ fun SignInScreen(
...
-                    text = stringResource(R.string.register),
-                    style = typography.text1.regular,
-                    color = colors.primary.main
-                )
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    horizontalArrangement = Arrangement.Center,
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
```
Comment (volodymyr-x) [inline app/src/main/java/com/mr/nemo/dragonfly/ui/screen/auth/signin/SignInScreen.kt]: @bohdanzdaniekin Why do you use a Column with only one Row inside?

**Review by slothdeboss — APPROVED** [review]

**Review by volodymyr-x — APPROVED** [review]

File: app/src/main/java/com/mr/nemo/dragonfly/ui/screen/auth/signin/SignInScreen.kt (hunk 2 above)
Comment (bohdanzdaniekin) [reply app/src/main/java/com/mr/nemo/dragonfly/ui/screen/auth/signin/SignInScreen.kt]: @igel911 Yeah, seems like we can use Box instead, fixed

**Pull request merged**